compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SingleAbstractMethodLowering.kt

```diff
@@ -180,7 +189,7 @@ class SingleAbstractMethodLowering(val context: CommonBackendContext) : ClassLow
             // common, as `Interface { something }` is a local function + a SAM-conversion of a reference
             // to it into an implementation.
             private fun createFunctionProxy(superType: IrType, reference: IrFunctionReference): IrClass =
-                implement(superType, reference.arguments.mapNotNull { it?.type }) { fields ->
+                implement(superType, reference.arguments.mapNotNull { it?.type }, reference.symbol.owner) { fields ->
```

Comment:
This would also copy annotations from non-lambdas (e.g. in `f(::g)`, `f` would see `g`'s annotations), and the non-IR backend does not do that.

Reply:
Thanks! Yes, extending the tests show that this is indeed the case. I'll have a look if I can replicate the behavior of the old backend here.

Reply:
Added more tests and implemented this in `createFunctionProxyInstance` for lambdas and anonymous functions only. That seems to match the behavior of the old backend.

@pyos, can you have another look?